tests/functional/base.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (C) 2017  Custodia Project Contributors - see LICENSE file
+from __future__ import absolute_import
+
+import os
+import shutil
+import socket
+import subprocess
+import sys
+import time
+from string import Template
+
+import pytest
+
+from custodia.client import CustodiaHTTPClient
+from custodia.compat import url_escape
+from custodia.server.args import parse_args
+from custodia.server.config import parse_config
+
+
+CONFIGURATION = u"""
+[DEFAULT]
+logdir = ${TEST_DIR}
+libdir = ${TEST_DIR}
+rundir = ${TEST_DIR}
+socketdir = ${TEST_DIR}
+
+[global]
+server_socket = ${TEST_DIR}/custodia.sock
+auditlog = ${TEST_DIR}/custodia.audit.log
+server_string = Test_Custodia_Server
+debug = false
+
+# Accepts any request that specifies an arbitrary REMOTE_USER header
+[auth:header]
+handler = SimpleHeaderAuth
+header = REMOTE_USER
+
+# Allow requests for all paths under '/' and '/secrets/'
+[authz:paths]
+handler = SimplePathAuthz
+paths = / /secrets/
+
+# Store secrets in a sqlite database called custodia.db in the table 'secrets'
+[store:simple]
+handler = SqliteStore
+dburi = ${TEST_DIR}/custodia.db
+table = secrets
+
+# Serve starting from '/' and using the 'simple' store and the 'Root' handler
+[/]
+handler = Root
+store = simple
+"""
+
+
+class FunctionalTest(object):
```

Comment:
I'd rather a better name for this class, Functional Test is a kind of test, this is just a base class which provides a Custodia server